ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java

```diff
@@ -80,10 +81,25 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
 
     final Object requestId = requestContext.getRequest().getId();
 
-    final EngineForkchoiceUpdatedParameter forkChoice =
-        requestContext.getRequiredParameter(0, EngineForkchoiceUpdatedParameter.class);
-    final Optional<EnginePayloadAttributesParameter> maybePayloadAttributes =
-        requestContext.getOptionalParameter(1, EnginePayloadAttributesParameter.class);
+    final EngineForkchoiceUpdatedParameter forkChoice;
+    try {
+      forkChoice = requestContext.getRequiredParameter(0, EngineForkchoiceUpdatedParameter.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+      throw new InvalidJsonRpcParameters(
+          "Invalid engine forkchoice updated parameter (index 0)",
+          RpcErrorType.INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS,
+          e);
+    }
+    final Optional<EnginePayloadAttributesParameter> maybePayloadAttributes;
+    try {
+      maybePayloadAttributes =
+          requestContext.getOptionalParameter(1, EnginePayloadAttributesParameter.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+      throw new InvalidJsonRpcParameters(
+          "Invalid engine payload attributes parameter (index 1)",
+          RpcErrorType.INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMS,
```

Comment:
This name is similar to `INVALID_ENGINE_PAYLOAD_PARAMS` used in the new payload and it's not clear that this is a engine forkchoice updated parameter that is failing. Don't think we need a new separate error type and can reuse `INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS`. Will the Exception message `Invalid engine payload attributes parameter (index 1)` show somewhere so we can distinguish which parameter failed?

Reply:
The exception message will only show up in logs. The message returned in the rpc response will be the one specified in the RpcErrorType. I named it INVALID_ENGINER_PAYLOAD_ATTRIBUTES_PARAMS due to the type of object being extracted from the request (in this case `EnginePayloadAttributesParameter`). Reusing INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS in this location would mean both parameters required in AbstractEngineForkchoiceUpdated would give the same error message.

Reply:
Ok, I'll make that update now